services/wallet/price.go

```diff
@@ -0,0 +1,82 @@
+package wallet
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+)
+
+type PricesPerTokenAndCurrency = map[string]map[string]float64
+
+type PriceManager struct {
+	db *sql.DB
+}
+
+func NewPriceManager(db *sql.DB) *PriceManager {
+	return &PriceManager{db: db}
+}
+
+func (pm *PriceManager) FetchPrices(symbols []string, currencies []string) (PricesPerTokenAndCurrency, error) {
+	result, err := fetchCryptoComparePrices(symbols, currencies)
+	if err != nil {
+		return result, err
+	}
+	err = pm.updatePriceCache(result)
+	return result, err
+}
+
+func (pm *PriceManager) updatePriceCache(prices PricesPerTokenAndCurrency) error {
+	insert, err := pm.db.Prepare(`INSERT OR REPLACE INTO price_cache 
+		(token,	currency, price)
+		VALUES
+		(?, ?, ?)`)
+	if err != nil {
+		return err
+	}
+
+	for token, pricesPerCurrency := range prices {
+		for currency, price := range pricesPerCurrency {
+			_, err = insert.Exec(token, currency, fmt.Sprintf("%f", price))
```

Comment:
@dlipicar for multiple actions in one go like this one we should use begin/commit transaction. Could you update it?

Reply:
I'll look that up. Sorry, don't have much experience with SQL